src/Azure.Functions.Cli/Actions/HostActions/StartHostAction.cs

```diff
@@ -480,6 +514,128 @@ public override async Task RunAsync()
             await runTask;
         }
 
+        private async Task<bool> ShouldStartChildProcessFromHostRuntime(bool isCurrentProcessNet8Build, bool isVerbose)
```

Comment:
Can we please have a unit test for this logic covering all the expected scenarios? I think that will help highlight some opportunities for simplification and some gaps.

Removing the, no longer needed, `isCurrentProcessNet8Build` argument will help identify how this may be simplified.

Reply:
Added unit tests for all the happy paths (to check if the func.exe is located in the correct directory)

Reply:
Added all the unit tests now (for not happy path as well)